ztools/x509/json.go

```diff
@@ -228,37 +229,41 @@ func (c *Certificate) MarshalJSON() ([]byte, error) {
 	jc.SubjectKeyInfo.KeyAlgorithm = c.PublicKeyAlgorithm
 
 	// Include all subject names, DNS names there are
-	for _, obj := range c.Subject.Names {
-
-		switch name := obj.Value.(type) {
-		case string:
-
-			flag := false
-
-			if len(name) > 2 && name[0] == '*' {
-				flag = govalidator.IsURL(name[2:])
-			} else {
-				flag = govalidator.IsURL(name)
-			}
+	name := c.Subject.CommonName
+	isValid := false
+
+	if len(name) > 2 && name[0] == '*' {
```

Comment:
Right. So should we consider that valid or not? 

@dadrian's right that I should look at names[1], there could be a weird case of *abc.blah.com, where this code would check c.blah.com as a result, which is *technically* valid but meaningless. I'll make this more robust.